File: app/src/main/java/laurens/hearthstoneassessment/model/Card.kt
package laurens.hearthstoneassessment.model

import android.arch.persistence.room.Entity
import android.arch.persistence.room.Ignore
import android.arch.persistence.room.PrimaryKey

@Entity(tableName = "card")
data class Card
@JvmOverloads
constructor(
    @PrimaryKey val cardId: String,
    val name: String,
    val cardSet: String,
    val type: String,
    val locale: String,
    val text: String?,
    val rarity: String?,
    val cost: String?,
    val attack: String?,
    val health: String?,
    val flavor: String?,
    val faction: String?,
    val durability: String?,
    val race: String?,
    val artist: String?,
    val multiClassGroup: String?,
    val playerClass: String?,
    val howToGet: String?,
    val howToGetGold: String?,
    val img: String?,
    val imgGold: String?,
    val collectible: Boolean?,
    val elite: Boolean?,
    @Ignore val classes: Collection<String> = emptyList(),
    @Ignore val mechanics: Collection<Mechanic> = emptyList()
)

File: HearthstoneHelper/HearthstoneHelper/Services/Entities/Card.swift
//
// Created by Maxim Berezhnoy on 13/12/2019.
// Licensed under the MIT license
//
// Copyright (c) 2019 rencevio. All rights reserved.

import Foundation

struct Card: Decodable {
    typealias ID = String
    
    struct Mechanic: Decodable {
        let name: String
    }
    
    let cardId: ID
    let name: String
    let type: String
    let rarity: String?
    let mechanics: [Mechanic]?
    let img: URL?
    let playerClass: String?
    let flavor: String?
}